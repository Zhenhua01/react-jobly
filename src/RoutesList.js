import { Route, Routes, Navigate } from "react-router-dom";
import HomePage from './Home/HomePage';
import CompanyList from './Company/CompanyList';
import JobList from './Job/JobList';
import CompanyDetail from './Company/CompanyDetail';
import SignupForm from './Form/SignupForm';
import LoginForm from './Form/LoginForm';
import ProfileForm from './Form/ProfileForm';

/** Routes for Jobly */

function RoutesList({ signup, login, update }) {

  return (
    <div className="RoutesList">

      <Routes>
        <Route path='*' element={<Navigate to='/' />} />
        <Route path='/' element={<HomePage />} />

        <Route path='/companies' element={<CompanyList />} />
        <Route path='/jobs' element={<JobList />} />
        <Route path='/companies/:handle' element={<CompanyDetail />} />

        <Route path='/signup' element={<SignupForm signup={signup} />} />
        <Route path='/login' element={<LoginForm login={login} />} />
        <Route path='/profile' element={<ProfileForm update={update} />} />
      </Routes>

    </div>
  );
}

export default RoutesList;