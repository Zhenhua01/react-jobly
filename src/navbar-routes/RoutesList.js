import { Route, Routes, Navigate } from "react-router-dom";
import HomePage from '../homepage/HomePage';
import CompanyList from '../companies/CompanyList';
import JobList from '../jobs/JobList';
import CompanyDetail from '../companies/CompanyDetail';
import SignupForm from '../forms/SignupForm';
import LoginForm from '../forms/LoginForm';
import ProfileForm from '../forms/ProfileForm';
import { useContext } from "react";
import userContext from '../context/userContext';


/** Routes list for Jobly App. */

function RoutesList({ signup, login, update }) {
  const { user } = useContext(userContext);

  return (
    <div className="RoutesList">

      <Routes>
        <Route path='*' element={<Navigate to='/' />} />
        <Route path='/' element={<HomePage />} />

        {user &&
          <>
            <Route path='/companies' element={<CompanyList />} />
            <Route path='/companies/:handle' element={<CompanyDetail />} />
            <Route path='/jobs' element={<JobList />} />
            <Route path='/profile' element={<ProfileForm update={update} />} />
          </>
        }

        <Route path='/login' element={<LoginForm login={login} />} />
        <Route path='/signup' element={<SignupForm signup={signup} />} />
      </Routes>

    </div>
  );
}

export default RoutesList;