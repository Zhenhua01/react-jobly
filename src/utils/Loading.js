import React from "react";
import "./Loading.css";


/** Loading message to display when making API calls for data. */

function Loading() {
  return (
    <div className="Loading">
      Loading ...
    </div>
  );
}

export default Loading;