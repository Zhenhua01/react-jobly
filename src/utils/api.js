import axios from "axios";

// Use port 3001 when connecting to local backend server
// const BASE_URL = process.env.REACT_APP_BASE_URL || "http://localhost:3001";

// Use heroku deployed backend when local server not available
const BASE_URL =
    process.env.REACT_APP_BASE_URL || "https://jobly-zhl-exp.herokuapp.com";

/** API Class.
 *
 * Static class tying together methods used to get/send to to the API.
 * There shouldn't be any frontend-specific stuff here, and there shouldn't
 * be any API-aware stuff elsewhere in the frontend.
 */

 class JoblyApi {

  static token = "";

  static async request(endpoint, data = {}, method = "get") {
    console.debug("API Call:", endpoint, data, method);

    const url = `${BASE_URL}/${endpoint}`;
    const headers = { Authorization: `Bearer ${JoblyApi.token}` };
    const params = (method === "get")
        ? data
        : {};

    try {
      return (await axios({ url, method, data, params, headers })).data;
    } catch (err) {
      console.error("API Error:", err.response);
      let message = err.response.data.error.message;
      throw Array.isArray(message) ? message : [message];
    }
  }

  // Individual API routes

  /** Get details on a company by handle. */

  static async getCompany(handle) {
    let res = await this.request(`companies/${handle}`);
    return res.company;
  }

  /** Get all companies or companies matching search term. */

  static async getCompanies(data = {}) {
    let res = await this.request(`companies`, data);
    return res.companies;
  }

  /** Get all jobs or jobs matching search term. */

  static async getJobs(data = {}) {
    let res = await this.request(`jobs`, data);
    return res.jobs;
  }

  /** Registers a new user. */

  static async signup(data) {
    let res = await this.request(`auth/register`, data, "post");
    return res.token;
  }

  /** Logins user. */

  static async login(data) {
    let res = await this.request(`auth/token`, data, "post");
    return res.token;
  }

  /** Update user information. */

  static async update(username, data) {
    let res = await this.request(`users/${username}`, data, "patch");
    return res.user;
  }

  /**Get user information. */

  static async getUserInfo(username) {
    let res = await this.request(`users/${username}`);
    return res.user;
  }

  /**Add job to user applications. */

  static async applyToJob(username, id) {
    await this.request(`users/${username}/jobs/${id}`, {}, "post");
    let res = await this.request(`users/${username}`);
    return res.user;
  }
}


export default JoblyApi;