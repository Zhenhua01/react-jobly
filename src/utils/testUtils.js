import React from "react";
import userContext from "../context/userContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
  applications: [],
};

const UserProvider = ({
  children,
  user = demoUser,
  hasAppliedToJob = () => false,
}) => (
  <userContext.Provider value={{ user, hasAppliedToJob }}>
    {children}
  </userContext.Provider>
);

export { UserProvider };
