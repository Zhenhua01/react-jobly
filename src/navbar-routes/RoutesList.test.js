import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import RoutesList from "./RoutesList";
import { UserProvider } from "../utils/testUtils";

it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <RoutesList />
        </UserProvider>
      </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <RoutesList />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
