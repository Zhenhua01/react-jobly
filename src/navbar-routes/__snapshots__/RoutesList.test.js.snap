// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="RoutesList"
  >
    <div
      class="HomePage"
    >
      <h1>
        Jobly
      </h1>
      <p>
        All the jobs in one, convenient place.
      </p>
      <p>
        Welcome back !
      </p>
    </div>
  </div>
</DocumentFragment>
`;
